Packages/io.chainsafe.web3-unity.web3auth/Samples~/Web3.Unity Web3Auth/Scripts/UI/Web3AuthLoginScreen.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class Web3AuthLoginScreen : LoginScreen
+{
+    /// <summary>
+    /// Struct used for pairing login buttons to Web3 auth providers.
+    /// Used when adding <see cref="Web3AuthLogin.LoginWithWeb3Auth"/> as listeners to the buttons.
+    /// </summary>
+    [Serializable]
+    public struct ProviderAndButtonPair
+    {
+        public Button Button;
+        public Provider Provider;
+    }
+    
+    [SerializeField] private List<ProviderAndButtonPair> providerAndButtonPairs;
+
+    public Web3AuthLogin Web3AuthLogin
```

Comment:
Why not make the LoginScreen to be an actual Generic parameter, like in the singleton pattern? And Web3AuthLogin would implement LoginScreen<Web3AuthLoginScreen>

Reply:
Not really, since Web3AuthLogin inherits Login and Web3AuthLoginScreen : LoginScreen which in turn implements Screen. A possible solution would be to have Screen be an interface instead and have LoginScreen inherit Login and implement IScreen that way all Logins can just inherit from LoginScreen, I'll try it out and see if it's feasible.